features/junit_format.feature

```diff
@@ -1,96 +1,18 @@
   Feature: JUnit Formatter
-  In order integrate with other development tools
+  In order to integrate with other development tools
   As a developer
   I need to be able to generate a JUnit-compatible report
 
-  Scenario: Normal Scenario's
-    Given a file named "features/bootstrap/FeatureContext.php" with:
-      """
-      <?php
-
-      use Behat\Behat\Context\Context,
-          Behat\Behat\Tester\Exception\PendingException;
-
-      class FeatureContext implements Context
-      {
-          private $value;
-
-          /**
-           * @Given /I have entered (\d+)/
-           */
-          public function iHaveEntered($num) {
-              $this->value = $num;
-          }
-
-          /**
-           * @Then /I must have (\d+)/
-           */
-          public function iMustHave($num) {
-              PHPUnit\Framework\Assert::assertEquals($num, $this->value);
-          }
-
-          /**
-           * @When /I add (\d+)/
-           */
-          public function iAdd($num) {
-              $this->value += $num;
-          }
-
-          /**
-           * @When /^Something not done yet$/
-           */
-          public function somethingNotDoneYet() {
-              throw new PendingException();
-          }
-      }
-      """
-    And a file named "features/World.feature" with:
-      """
-      Feature: World consistency
-        In order to maintain stable behaviors
-        As a features developer
-        I want, that "World" flushes between scenarios
-
-        Background:
-          Given I have entered 10
+  Background:
+    Given I set the working directory to the "JunitFormat" fixtures folder
 
-        Scenario: Undefined
-          Then I must have 10
-          And Something new
-          Then I must have 10
-
-        Scenario: Pending
-          Then I must have 10
-          And Something not done yet
-          Then I must have 10
-
-        Scenario: Failed
-          When I add 4
-          Then I must have 13
-
-        Scenario Outline: Passed & Failed
-          When I add <value>
-          Then I must have <result>
-
-          Examples:
-            | value | result |
-            |  5    | 16     |
-            |  10   | 20     |
-            |  23   | 32     |
-
-      Scenario Outline: Another Outline
-        When I add <value>
-        Then I must have <result>
-
-        Examples:
-          | value | result |
-          | 5     | 15     |
-          | 10    | 20     |
-      """
-    When I run "behat --no-colors -f junit -o junit --snippets-for=FeatureContext --snippets-type=regex"
+  Scenario: Run a single feature
+    Given I want to run the suite "single_feature"
+    And I set the output folder to the system temp folder
+    And I run "behat --no-colors -f junit --snippets-for=SingleFeatureContext --snippets-type=regex"
```

Comment:
I feel this is a bit hard to follow, because the final line looks like it is an explicit `behat` command invocation but in fact it is merged in with the parameters set above. I feel like splitting `I want to run the suite "single_feature"` out to a separate step makes this less obvious, not more.

I think maybe there's a few things that might help:

* Move more of the "common" CLI arguments to the suite config file - e.g. the formatter to use, maybe also the output path?
* Have a step like `Given I will provide the following arguments for all behat invocations:` as a TableNode in e.g. the Background with options that are required to make the output match but not critical to the feature we're testing?
* Change the `I run` to be `I run behat with the following additional options:` as a TableNode so that the individual arguments are on separate lines

So here we might have:

```gherkin
Given I will provide the following arguments for all behat invocations:
  # not sure the best way to represent flags in a table / array we can merge, but {SET}
  # to add it to the list of options and {NOT SET} to override? Or maybe it's safer if
  # we don't allow these values to be overridden later in the feature...
  | --no-colors     | {SET} | 
  | --snippets-type | regex |

# ... 
When I run behat with the following additional options:
  | --snippets-for | SingleFeatureContext |
  | --suite        | single_feature      |
  # not sure about this, but want to avoid a separate "I set the output folder to ... step"
  | --out          | {SYSTEM_TMP_DIR}     |